Tests/Tests/TDMiscTests.m

```diff
@@ -75,5 +75,164 @@ - (void)testEscapeID
+    STAssertEqualObjects(cleanString, @"myscheme://*****@myhost.com:1234/db?all_docs=true",
+                         @"not cleaned: %@", cleanString);
+    
+    testUrl = [NSURL URLWithString:@"http://adam:adamspassword@myhost.com"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"http://*****@myhost.com", @"not clean: %@", cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam:adamspasswordmyhost@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://adam@myhost.com/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://*****@myhost.com/db", @"not clean: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://nothing.to.clean/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://nothing.to.clean/db", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://@/db", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://:@"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://:@", @"should not have changed: %@",
+                         cleanString);
+    
+    testUrl = [NSURL URLWithString:@"https://:@/db"];
+    cleanString = TDCleanURLtoString(testUrl);
+    STAssertEqualObjects(cleanString, @"https://:@/db", @"should not have changed: %@",
+                         cleanString);
+    
+    
+    
+    NSString *errorMessage = @"NSURL did not have both a scheme and host. Suppressing string value for security.";
+    
+    //
+    //these should fail due to typo in the '://', which prevents NSURL from parsing the string.
+    //
+    testUrl = [NSURL URLWithString:@"https:/adam:adamspassword@myhost.com/db"];
+    STAssertEqualObjects(testUrl.scheme, @"https", @"scheme not https?  %@", testUrl.scheme);
+    STAssertNil(testUrl.host, @"host not nil?  %@", testUrl.host);
+    STAssertNil(testUrl.user, @"user not nil?  %@", testUrl.user);
+    STAssertNil(testUrl.password, @"password not nil?  %@", testUrl.password);
```

Comment:
Well, it's actually testing NSURL's parsing behavior, which doesn't seem to be very well documented to me. It only refers you to the RFC 1808 spec to say when it returns nil because of a formatting issue. I'm not testing that my hardcoded values exist -- I'm making sure that NSURL doesn't parse these particular examples, as expected. So, I put them there as a sanity check and, in a way, some kind of proof and documentation of the logic behind TDCleanURLtoString. 

But yeah, I can remove them if you still think they are overkill. And it's obviously not an exhaustive list of different typos that could occur